Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -85,15 +89,39 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, R
                     ->andWhere(sprintf('o.%1$s LIKE :%1$s', $property))
                     ->setParameter($property, $partial ? sprintf('%%%s%%', $value) : $value)
                 ;
-            } elseif ($metadata->isSingleValuedAssociation($property)
-                || $metadata->isCollectionValuedAssociation($property)
-            ) {
+            } elseif ($metadata->isSingleValuedAssociation($property)) {
+                if (!is_string($value)) {
+                    continue;
+                }
+
                 $value = $this->getFilterValueFromUrl($value);
 
                 $queryBuilder
                     ->join(sprintf('o.%s', $property), $property)
                     ->andWhere(sprintf('%1$s.id = :%1$s', $property))
-                    ->setParameter($property, $partial ? sprintf('%%%s%%', $value) : $value)
+                    ->setParameter($property, $value)
+                ;
+            } elseif ($metadata->isCollectionValuedAssociation($property)) {
+                $values = &$value;
```

Comment:
I just thought the naming would be clearer this way... What's the drawback here?